src/mudlet-lua/lua/geyser/GeyserButton.lua

```diff
@@ -0,0 +1,244 @@
+-- @field clickCommand The command to send when the button is clicked. For two state buttons, used when clicking in the 'up' state. Will be skipped if clickFunction is defined.
+-- @field clickFunction Optional function to run when button clicked. For two state buttons, used when clicking in the 'up' state. If defined takes precedence over clickCommand.
+-- @field downCommand The command to send when the two state button is clicked while in the 'down' state.
+-- @field downFunction The function to run when the two state button is clicked while in the 'down' state.
+-- @field downColor The color to make the button while it is in the 'down' state. Superceded by downStyle
+-- @field downStyle The stylesheet to use for the button while it is in the 'down' state. Supercedes downColor
+-- @field color The color to make the button. If a two state button, will be used for the 'up' state. Superceded by style
+-- @field style The stylesheet to use for the button. If a two state button, will be used for the 'up' state. Supercedes color
+-- @field msg The text to put on the button. If a two state button, will be used for the 'up' state.
+-- @field downMsg The text to put on the button in the 'down' state.
+Geyser.Button = {
+  name = "GeyserButtonClass",
+  height = 50,
+  width = 50,
+  downColor = "dark_green",
+  downMsg = "<center>Quicklook</center>",
+  downTooltip = "Click to quicklook",
+  downCommand = "quicklook",
+  color = "blue",
+  msg = "<center>Look</center>",
+  clickCommand = "look",
+  twoState = false,
+  state = "up",
+  toolTipDuration = 5,
+  tooltip = "Click to look",
+}
+Geyser.Button.parent = Geyser.Label
+setmetatable(Geyser.Button, Geyser.Label)
+Geyser.Button.__index = Geyser.Button
+
+--- Creates a new button and returns it
+-- @param cons table of constraints which define the button
+-- @param container the container to place the button into
+function Geyser.Button:new(cons, container)
+  cons = cons or {}
+  local consType = type(cons)
+  if consType ~= "table" then
+    printError(f"bad argument #1 type (cons as table expects, got {consType})", true, true)
+  end
+  cons.name = cons.name or Geyser.nameGen("button")
+  local me = self.parent:new(cons, container)
+  setmetatable(me, self)
+  me:setClickCallback(function() me:press() end)
+  me:setState(me.state)
+  me:resize() -- to pick up the Geyser.Button default size rather than Geyser.Label's
+  return me
+end
+
+--- Sets the state of the button to "up" or "down" and performs all accompanying visual transformations.
+-- Does not simulate the actual click, see Geyser.Button:press for that
+-- @param state the state to set the button to. One of "up" or "down"
+-- @see press
+function Geyser.Button:setState(state)
+  local stateType = type(state)
+  if stateType ~= "string" then
+    return nil, f"bad argument #1 type (state as string expected, got {stateType})"
+  end
+  state = state:lower()
```

Comment:
I don't think we do lowercasing in the C++ of explicit arguments like this and it might be better to stay consistent here, what do you think?

Reply:
I don't think it's very friendly to puke if it's 'Up' or 'UP' instead of 'up' though? Easier to lowercase the state and then check against that than to check against every case combination imo.

Reply:
It is case-sensitive everywhere they look already, they have to learn one way or another 😅  it's not a biggie though.